connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -573,9 +582,11 @@ public Void call() throws Exception {
                     return null;
                 }
 
-                if (worker.ownsConnector(connName)) {
+                if (assignment.connectors().contains(connName)) {
```

Comment:
In general the reason for handling exceptions in `DistributedHerder` rather than `Worker` - I figure the current behavior works well for `StandaloneHerder` which calls the same methods on `Worker`, since it should just crash if it runs into basic errors with the connector or its tasks.